pkg/operator/ceph/cluster/rbd/controller.go

```diff
@@ -166,23 +180,31 @@ func (r *ReconcileCephRBDMirror) reconcile(request reconcile.Request) (reconcile
 	// Always populate it during each reconcile
 	r.clusterInfo, _, _, err = mon.LoadClusterInfo(r.context, request.NamespacedName.Namespace)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrap(err, "failed to populate cluster info")
+		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to populate cluster info")
 	}
 
 	// Populate CephVersion
 	daemon := string(opconfig.MonType)
 	currentCephVersion, err := cephclient.LeastUptodateDaemonVersion(r.context, r.clusterInfo, daemon)
 	if err != nil {
-		return reconcile.Result{}, errors.Wrapf(err, "failed to retrieve current ceph %q version", daemon)
+		return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to retrieve current ceph %q version", daemon)
 	}
 	r.clusterInfo.CephVersion = currentCephVersion
 
+	// Add bootstrap peer if any
+	logger.Debug("reconciling ceph rbd mirror peers addition")
+	reconcileResponse, err = r.reconcileAddBoostrapPeer(cephRBDMirror, request.NamespacedName)
+	if err != nil {
+		updateStatus(r.client, request.NamespacedName, k8sutil.FailedStatus)
```

Comment:
How about calling `updateStatus()` for the failed case from the `reconcile()` method whenever `err != nil`? Currently some error code paths are not reporting an error here, and that way you could also avoid duplication of the `updateStatus()` call.